pcdsdevices/epics_motor.py

```diff
@@ -221,6 +186,80 @@ def check_limit_switches(self):
         else:
             return "Low [] High []"
 
+    def get_low_limit(self):
+        """Get the low limit."""
+        return self.low_limit_travel.get()
+
+    def set_low_limit(self, value):
+        """
+        Set the low limit.
+
+        Parameters
+        ----------
+        value : float
+            Limit of travel in the negative direction.
+
+        Raises
+        ------
+        ValueError
+            When motor in motion or position outside of limit.
+        """
+        if self.moving:
+            raise ValueError('Motor is in motion, cannot set the low limit!')
+
+        if value > self.position:
+            raise ValueError(f'Could not set motor low limit to {value} at'
+                             f' position {self.position}. Low limit must '
+                             'be lower than the current position.')
+
+        _current_high_limit = self._get_epics_limits()[1]
+        if value > _current_high_limit:
+            raise ValueError(f'Could not set motor low limit to {value}.'
+                             'Low limit must be lower than the current'
+                             f'high limit: {_current_high_limit}')
+
+        # update EPICS limits
+        self.low_limit_travel.put(value)
+
+    def get_high_limit(self):
+        """Get high limit."""
+        return self.high_limit_travel.get()
+
+    def set_high_limit(self, value):
+        """
+        Limit of travel in the positive direction.
+
+        Parameters
+        ----------
+        value : float
+            High limit value to be set.
+
+        Raises
+        ------
+        ValueError
+            When motor in motion or position outside of limit.
+        """
+        if self.moving:
+            raise ValueError('Motor is in motion, cannot set the high limit!')
+
+        if value < self.position:
+            raise ValueError(f'Could not set motor high limit to {value} '
+                             f'at position {self.position}. High limit '
+                             'must be higher than the current position.')
+
+        _current_low_limit = self._get_epics_limits()[0]
+        if value < _current_low_limit:
+            raise ValueError(f'Could not set motor high limit to {value}. '
+                             'High limit must be higher than the current low '
+                             f'limit: {_current_low_limit}')
+        # update EPICS limits
+        self.high_limit_travel.put(value)
+
+    def clear_limits(self):
+        """Set both low and high limits to 0."""
+        self.set_low_limit(0)
+        self.set_high_limit(0)
```

Comment:
This could fail if the limits are something like (-100, -50) before the function call, since the low limit can't be zero if the high limit is -50 due to the checks above. Funnily enough, `self.set_lim(0, 0)` also fails for similar reasons.

Might be easiest just to do straight puts to the PVs.

Reply:
Ok i changed it in 175f3e1 using the ` self.high_limit_travel.put(0)` and `self.low_limit_travel.put(0)` - i think that's what you meant. 

Thank you